state/execution.go

```diff
@@ -450,12 +459,23 @@ func (blockExec *BlockExecutor) asyncUpdateMempool(
 ) {
 	defer unlockMempool()
 
+	recheck := true
```

Comment:
We know the peer height, from the same already existing database, in the mempool protocol.

I don't see a point to the block executor to interact with the p2p layer.

Reply:
Same as above. Now I'm getting the peer info from the mempool reactor.